### PR TITLE
fix(public): drop Place bullet + fix dark-mode <code> on public DoC

### PR DESCRIPTION
**Pull request by aurangzaib048**
## Summary

Two related fixes to the public Declaration of Conformity page on the trust center, both surfaced from the same Slack thread:

1. **Place leak** — Annex V Section 8 "Place" rendered the operator's full signing location (e.g. "Naberezhnye Chelny, Russian Federation"). Public visitors don't need this; the manufacturer typically doesn't want it broadcast.
2. **Dark-mode `<code>` invisible** — inline code blocks in the DoC (e.g. `sboms/*.cdx.json`, `security.txt`) rendered with a near-white background in trust-center dark mode, leaving the text unreadable.

### Place — fix

- New `remove_signature_place_for_public` helper in `_public_helpers.py` strips the entire `- **Place:** …` bullet from the markdown before render.
- `ProductDoCPublicView.get` applies it between `fetch_doc_from_s3` and `markdown_to_html`.
- Regex terminator is `(?:\n|$)` so a Place bullet at end-of-file is also removed.
- **Stored markdown is unchanged.** PDF render + bundle export read directly from the same S3 blob, so the auditor / notified body still sees the real Place value — Annex V Section 8 compliance preserved.
- Initial iteration rendered `Place: _Redacted_`; reverted to dropping the bullet entirely (the placeholder added no signal).

### Dark-mode `<code>` — fix

- Tailwind's `prose` plugin gives inline `<code>` a light slate background. The `dark:prose-invert` class on the public templates would flip it, but Tailwind's `dark:` variant fires from the `.dark` class — and the trust-center theme system uses `data-theme="dark"` only (no class). So the dark inversion was never applied.
- Add a scoped CSS rule in `public_base.htmx.j2` keyed off `html[data-theme="dark"] .prose code`, using `rgb(var(--color-surface-elevated))` + `rgb(var(--color-text))` (the same CSS variables the surrounding dark block defines, so palette tweaks flow through automatically).
- Affects every public template that uses `.prose` (DoC + VDP).

### Files changed

- `sbomify/apps/compliance/views/_public_helpers.py` — `remove_signature_place_for_public()`
- `sbomify/apps/compliance/views/doc_public.py` — wire-up
- `sbomify/apps/compliance/tests/test_public_helpers.py` — new file, 8 helper unit tests (incl. EOF edge case)
- `sbomify/apps/compliance/tests/test_doc_public.py` — view integration test
- `sbomify/apps/core/templates/core/public_base.htmx.j2` — dark-mode `<code>` override

### Local verification

| Scenario | Result |
| --- | --- |
| Public DoC dark mode: Place bullet | gone — no `Place:` mention anywhere in body |
| Public DoC dark mode: Date / Name / Function / Signature | all intact |
| Public DoC dark mode: `<code>` blocks (Section 8) | `bgColor = rgb(51, 65, 85)` (slate-700), `textColor = rgb(248, 250, 252)` (slate-50) |
| Helper: unicode, idempotency, EOF, blank template, mid-paragraph inline `Place:` | 8/8 pass |
| Helper: regression for end-of-file Place bullet (no trailing `\n`) | passes |
| View integration: synthetic Naberezhnye/RU markdown → public render strips Place, keeps everything else | passes |

22 / 22 compliance tests pass.

## Test plan

- [ ] On stage: public DoC at `/compliance/public/product/<id>/declaration-of-conformity/` — no Place bullet, other Section 8 fields intact.
- [ ] Toggle trust-center to dark mode — inline `<code>` blocks (file paths in Section 8) render readable (light text on dark slate), not white-on-near-white.
- [ ] Download the DoC PDF / bundle export → real Place value still present (Annex V compliance preserved for the auditor / notified body).
- [ ] Existing public DoC tests still pass (`pytest sbomify/apps/compliance/tests/test_doc_public.py sbomify/apps/compliance/tests/test_public_helpers.py`).

### CODE REVIEW

File: sbomify/apps/compliance/views/_public_helpers.py [hunk 1]
```diff
@@ -208,6 +208,28 @@ def get_document_content(product: Any, document_kind: str, *, fresh_only: bool =
     return fetch_doc_from_s3(doc)
 
 
+# Matches the entire ``- **Place:** …\n`` bullet so we can drop it
+# from the public render. The PDF and bundle export hold the original
+# markdown — only the public trust-center view fetches it through this
+# helper. CRA Annex V Section 8 still requires Place to be present in
+# the authoritative DoC; the auditor / notified body sees the
+# unredacted value via the export ZIP.
+_DOC_PLACE_LINE_RE = re.compile(r"^- \*\*Place:\*\* .+\n", re.MULTILINE)
```
[the remark this answers is not in the document]
Comment (aurangzaib048) [reply sbomify/apps/compliance/views/_public_helpers.py]: Fixed in `5c1efb57`. Changed the regex terminator from `\\n` to `(?:\\n|$)` so a Place bullet on the final line (no trailing newline) is also stripped. Added a regression test `test_removes_place_when_it_is_the_last_line_without_trailing_newline` that fails without the fix and passes with it. Full suite still green (22/22).

File: sbomify/apps/core/templates/core/public_base.htmx.j2 [hunk 2]
```diff
@@ -247,6 +247,20 @@
...
+               dark mode. The `dark:prose-invert` class on doc_public /
+               vdp_public templates fixes this — but only when Tailwind's
+               `dark:` variant fires. Trust-center pages drive theme via
+               `data-theme="dark"` (their own state machine), so the variant
+               doesn't pick this up. Apply the dark-code background by hand
+               so the `<code>` text stays readable on the public DoC + VDP
+               pages until the variant unification lands. */
+            html[data-theme="dark"] .prose code {
+                background-color: rgb(51 65 85);
+                color: rgb(248 250 252);
```
[the remark this answers is not in the document]
Comment (aurangzaib048) [reply sbomify/apps/core/templates/core/public_base.htmx.j2]: Fixed in `5c1efb57`. The override now reads:

```css
html[data-theme="dark"] .prose code {
    background-color: rgb(var(--color-surface-elevated));
    color: rgb(var(--color-text));
}
```

Both variables are defined a few lines above in the same `html[data-theme="dark"]` block, so the override stays in sync with the rest of the public dark palette automatically. Browser-checked the resolved values are unchanged (`rgb(51, 65, 85)` bg, `rgb(248, 250, 252)` text).

File: sbomify/apps/compliance/views/doc_public.py [hunk 3]
```diff
@@ -77,6 +81,15 @@ def get(self, request: HttpRequest, product_id: str) -> HttpResponse:
...
             return HttpResponseNotFound("No Declaration of Conformity for this product")
 
+        # Strip Annex V Section 8 "Place" before rendering: the trust
+        # center is public, so the city / country where the operator
+        # signed shouldn't be broadcast. Dropping the bullet entirely
+        # (rather than rendering "[redacted]") keeps the public DoC
+        # focused on data we DO want to surface. The auditor-facing
+        # PDF + the bundle export keep the original markdown from S3
+        # unchanged, so Annex V Section 8 compliance is preserved.
+        content = remove_signature_place_for_public(content)
```
[the remark this answers is not in the document]
Comment (aurangzaib048) [reply sbomify/apps/compliance/views/doc_public.py:91]: Updated the PR title + body to reflect both actual behaviours:

- Place bullet is **dropped entirely** (initial draft used `_Redacted_` placeholder; reverted per follow-up feedback because the placeholder was noise). The view integration test in this PR (`test_signature_place_is_removed_in_public_render`) asserts "Place" doesn't appear in the public render at all.
- Dark-mode `<code>` fix is now documented in the summary + test plan.

The implementation and tests match the updated description; the previous mention of `_Redacted_` was stale text.

File: sbomify/apps/core/templates/core/public_base.htmx.j2 [hunk 4]
```diff
@@ -247,6 +247,21 @@
...
 
+            /* Tailwind's `prose` typography plugin gives inline `<code>` a
+               light slate background that turns into white-on-near-white in
+               dark mode. The `dark:prose-invert` class on doc_public /
+               vdp_public templates fixes this — but only when Tailwind's
+               `dark:` variant fires. Trust-center pages drive theme via
+               `data-theme="dark"` (their own state machine), so the variant
+               doesn't pick this up. Apply the dark-code background by hand
+               using the same dark-theme CSS variables defined just above
+               (re-use ``--color-surface-elevated`` and ``--color-text`` so
```
[the remark this answers is not in the document]
Comment (aurangzaib048) [reply sbomify/apps/core/templates/core/public_base.htmx.j2:258]: Both fixes (Place removal + dark-mode `<code>` override) are now documented in the updated PR title + summary + files-changed + test plan. The two issues surfaced on the same Slack thread and affect the same public DoC page, so bundling kept the round-trip short; happy to split into a separate PR if you'd prefer.

File: sbomify/apps/core/templates/core/public_base.htmx.j2 [hunk 5]
```diff
@@ -247,6 +247,21 @@
...
+               future palette tweaks flow through to <code> automatically). */
+            html[data-theme="dark"] .prose code {
+                background-color: rgb(var(--color-surface-elevated));
```
[the remark this answers is not in the document]
Comment (aurangzaib048) [reply sbomify/apps/core/templates/core/public_base.htmx.j2:261]: Already fixed in `5c1efb57` (one commit before this re-review) — the override now uses `rgb(var(--color-surface-elevated))` and `rgb(var(--color-text))` instead of literal RGB. The diff hunk Copilot is showing here is from the older `04f9a1a3` commit. Browser-checked the resolved values are unchanged after the variable swap.

**Review by vpetersson — APPROVED** [review]

**Review by vpetersson — CHANGES_REQUESTED** [review]
Note in Slack

**Review by vpetersson — APPROVED** [review]
